adding "Retry-After" (#79814)
Azure/azure-content: articles/connectors/connectors-native-http.md

<a name="disable-location-header-check"></a>

> ### Retry-After  

Reviewer: ```suggestion
### Set up interval between retry attempts with the Retry-After header
```

> The http action respects the retry after with 429 status code if received one .

Reviewer: After finding an existing work item about this update and reading the background, I'm editing this suggestion to the following revision plus adding a blank line between the heading and body text:

```suggestion

The HTTP action supports using the `Retry-After` header and the `429 - Too many requests` status code. To set up the interval between retry attempts, add the `Retry-After` header to the HTTP action, and set the value in seconds:
```

Reviewer: OK, the team clarified that `Retry-After` is a header value, so edited my suggestion.

Author: I am not sure for the http trigger

Reviewer: @mbarqawi: OK, we can remove the trigger if we're not sure. I'll update my revision suggestion.

> the value will be considered as Seconds

Reviewer: Delete this line as the suggestion includes this text.

```suggestion
the value will be considered as Seconds
```

Reviewer: @mbarqawi: This line still appears, "the value will be considered as Seconds". Please delete.

Reviewer: @mbarqawi: Please delete line 308, `the value will be considered as Seconds`, because the paragraph above already mentions that the value is in seconds.

> 

Reviewer: This example doesn't seem to follow the documented format. Will get back to you on this.

Reviewer: Here you go:

```json
"HTTP": {
   "inputs": {
      "headers": {
         "Retry-After": "300"
      },
      "method": "GET",
      "uri": https://my-target-url.com
   },
   "runAfter": {},
   "type": "Http"
}
```

Reviewer: @mbarqawi: I still see the original example, which doesn't provide enough context. Please use the fuller example instead, which shows how the HTTP action uses the header:

```json
"HTTP": { 
    "inputs": {
        "headers": {
            "Retry-After": "300"
        }, 
        "method": "GET", 
        "uri": https://my-target-url.com
    }, 
    "runAfter": {},
    "type": "Http" 
}
```

Reviewer: @mbarqawi: A reminder to replace the original example with the fuller example.

<a name="disable-location-header-check"></a>

### Set up interval between retry attempts with the Retry-After header

> The HTTP action supports using the `Retry-After` header and the `429 - Too many requests` and `202 - Accepted` response status codes. To set up the interval between retry attempts, add the `Retry-After` header to the HTTP action response, and set the value in seconds:

Reviewer: Grammar fix:

```suggestion
The HTTP action supports using the `Retry-After` header with the `429 - Too many requests` and `202 - Accepted` response status codes. To set up the interval between retry attempts, add the `Retry-After` header to the HTTP action response, and set the value in seconds:
```

> The HTTP action supports using the `Retry-After` header with the `429 - Too many requests` and `202 - Accepted` response status codes. To set up the interval between retry attempts, add the `Retry-After` header to the HTTP action response, and set the value in seconds:

Reviewer: @mbarqawi: I made more clarification edits:

```suggestion
To specify the number of seconds between retry attempts, you can add the `Retry-After` header to the HTTP action. For example, if the target endpoint returns the `429 - Too many requests` status code, you can specify a longer interval between retries. The `Retry-After` header also works with the `202 - Accepted` status code.
```

<a name="disable-location-header-check"></a>

### Set up interval between retry attempts with the Retry-After header

To specify the number of seconds between retry attempts, you can add the `Retry-After` header to the HTTP action response. For example, if the target endpoint returns the `429 - Too many requests` status code, you can specify a longer interval between retries. The `Retry-After` header also works with the `202 - Accepted` status code.

> Here is the same example that shows the HTTP action response that contains `Retry-After`

Reviewer: Add missing colon:

```suggestion
Here is the same example that shows the HTTP action response that contains `Retry-After`:
```
